blog/2023-07-31-quarterly-earnings-chatbot/index.md

```diff
@@ -0,0 +1,227 @@
+
+### Let's get started!
+
+<hr />
+
+## Step 1: Install python dependencies
+
+First make sure you have Prem running locally and set up the instances for Whisper Tiny, Weaviate, Vicuna, and All MiniLM. 
+
+Next, let's install some Python dependencies:
+
+```txt
+pip install langchain openai streamlit
+```
+
+## Step 2: Transcribe audio to text
+
+For this app, we'll be uploading an MP3 file containing the audio for Nvidia's earnings call using the Streamlit UI.
+
+```python
+import openai
+import os
+from langchain.embeddings import OpenAIEmbeddings
+from langchain.text_splitter import RecursiveCharacterTextSplitter
+from langchain.docstore.document import Document
+from langchain.vectorstores import Weaviate
+from langchain.chat_models import ChatOpenAI
+from langchain.schema import HumanMessage
+
+import tempfile
+import streamlit as st
+
+os.environ["OPENAI_API_KEY"] = "random-string"
+
+embeddings = OpenAIEmbeddings(openai_api_base="http://127.0.0.1:8444/v1")
+weaviate_url = "http://127.0.0.1:8080"
+
+def convert_audio_to_text(audio_file_path) -> str:
+    openai.api_base = "http://127.0.0.1:10111/v1"
+    openai.api_key = "random-string"
+
+    audio_file = open(audio_file_path, "rb")
+    transcript = openai.Audio.transcribe("whisper-1", audio_file)
+    return transcript.get('text')
+```
+
+The `convert_audio_to_text` function takes in a path to an audio file as an argument and then used the OpenAI Whisper Tiny model running locally to transcribe the audio to text.
+
+You can call the `convert_audio_to_text` function and give it the path to the mp3 file if you want to test out the function.
+
+## Step 3: Chunk the text
+
+Now that we have the text from the previous step we need to split it into multiple chunks. An earnings call is typically an hour long, so you don't want to process a giant text blob.
+
+Instead, if you can cut the text blob into multiple pieces, it becomes much easier to process and query.
+
+```python
+def create_chunks(text_blob: str):
+    text_splitter = RecursiveCharacterTextSplitter(
+        chunk_size=500,
+        chunk_overlap=20,
+        length_function=len
+    )
+
+    texts = text_splitter.create_documents([text_blob])
+    return texts
+```
+
+## Step 4: Load the chunks into Weaviate vector store
+
+```python
+def add_to_vectorstore(texts):
+    embeddings = OpenAIEmbeddings(openai_api_base="http://127.0.0.1:8444/v1")
+    documents = []
+    for t in texts:
+        doc = Document(page_content=t.page_content)
+        documents.append(doc)
+
+    vectorstore = Weaviate.from_documents(
+        documents,
+        embeddings,
+        weaviate_url=weaviate_url,
+        by_text=False,
+    )
+    return vectorstore
+```
+
+We want to store the chunked text so that we can query it later on. We can use Langchain's `Weaviate.from_documents` method to easily upload our text into our Weaviate vector database.
```

Comment:
cite docs whenever you mention a function

```suggestion
We want to store the chunked text so that we can query it later on. We can use Langchain's [`Weaviate.from_documents`](some link) method to easily upload our text into our Weaviate vector database.
```